apps/addons/tests/test_commands.py

```diff
@@ -52,3 +60,124 @@ def is_forced(ids, force):
         assert force
     monkeypatch.setattr(SIGN_ADDONS, is_forced)
     call_command('sign_addons', 123, force=True)
+
+
+# Test the "approve_addons" command.
+
+@pytest.mark.django_db
+def test_approve_addons_get_files_incomplete():
+    """An incomplete add-on can't be approved."""
+    addon = amo.tests.addon_factory(status=amo.STATUS_NULL)
+    assert approve_addons.get_files([addon.guid]) == []
+
+
+@pytest.mark.django_db
+def test_approve_addons_get_files_bad_guid():
+    """An add-on with another guid doesn't get approved."""
+    addon1 = amo.tests.addon_factory(status=amo.STATUS_UNREVIEWED, guid='foo')
+    addon1_file = addon1.latest_version.files.get()
+    addon1_file.update(status=amo.STATUS_UNREVIEWED)
+    # Create another add-on that we won't get the files for.
+    addon2 = amo.tests.addon_factory(status=amo.STATUS_UNREVIEWED, guid='bar')
+    addon2_file = addon2.latest_version.files.get()
+    addon2_file.update(status=amo.STATUS_UNREVIEWED)
+    # There's only the addon1's file returned, no other.
+    assert approve_addons.get_files(['foo']) == [addon1_file]
+
+
+@pytest.fixture(
+    params=[(amo.STATUS_UNREVIEWED, amo.STATUS_UNREVIEWED, 'prelim'),
+            (amo.STATUS_LITE, amo.STATUS_UNREVIEWED, 'prelim'),
+            (amo.STATUS_NOMINATED, amo.STATUS_NOMINATED, 'full'),
+            (amo.STATUS_PUBLIC, amo.STATUS_NOMINATED, 'full'),
+            (amo.STATUS_LITE_AND_NOMINATED, amo.STATUS_LITE, 'full')],
+    ids=['1-1-prelim',  # Those are used to build better test names, for the
+         '8-1-prelim',  # tests using this fixture, eg:
+         '3-3-full',    # > test_approve_addons_get_files[1-1-prelim]
+         '4-3-full',    # instead of simply:
+         '9-8-full'])   # > test_approve_addons_get_files[usecase0]
+def use_case(request, db):
+    """This fixture will return quadruples for different use cases.
+
+    Addon                   | File1 and 2        | Review type
+    ==============================================================
+    waiting for prelim      | waiting for prelim | prelim reviewed
+    prelim reviewed         | waiting for prelim | prelim reviewed
+    waiting for full        | waiting for full   | fully reviewed
+    fully reviewed          | waiting for full   | fully reviewed
+    prelim waiting for full | prelim reviewed    | fully reviewed
```

Comment:
Given the complexity of the thing (and the newness too, not sure many are accustomed to pytest fixtures), it's the least I could do. Happy you like it ;)